c tests fail to build

```
$ meson build
The Meson build system
Version: 0.54.3
Source dir: /home/grg/src/tskit/c
Build dir: /home/grg/src/tskit/c/build
Build type: native build
Project name: tskit
Project version: undefined
C compiler for the host machine: cc (gcc 10.1.0 "cc (GCC) 10.1.0")
C linker for the host machine: cc ld.bfd 2.34.0
C++ compiler for the host machine: c++ (gcc 10.1.0 "c++ (GCC) 10.1.0")
C++ linker for the host machine: c++ ld.bfd 2.34.0
Host machine cpu family: x86_64
Host machine cpu: x86_64

|Executing subproject kastore method meson
|
|Project name: kastore
|Project version: undefined
|C compiler for the host machine: cc (gcc 10.1.0 "cc (GCC) 10.1.0")
|C linker for the host machine: cc ld.bfd 2.34.0
|C++ compiler for the host machine: c++ (gcc 10.1.0 "c++ (GCC) 10.1.0")
|C++ linker for the host machine: c++ ld.bfd 2.34.0
|Build targets in project: 1
|Subproject kastore finished.

Library m found: YES
Found pkg-config: /usr/bin/pkg-config (1.7.3)
Run-time dependency cunit found: YES 2.1-3
Build targets in project: 21

tskit undefined

  Subprojects
    kastore: YES

Found ninja-1.10.0.git.kitware.jobserver-1 at /home/grg/.local/bin/ninja
Generating targets:   0%|                                                    | 0/21 [00:00<?,                                                                                               Writing build.ninja:   0%|                                                   | 0/82 [00:00<?, 
```

```
$ ninja -C build
ninja: Entering directory `build'
[32/55] Linking target test_core
FAILED: test_core
cc  -o test_core 'test_core@exe/tests_test_core.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group libtskit.a subprojects/kastore/libkastore.a libtestlib.a -lm /usr/lib/libcunit.so -Wl,--end-group '-Wl,-rpath,$ORIGIN/:$ORIGIN/subprojects/kastore' -Wl,-rpath-link,/home/grg/src/tskit/c/build/ -Wl,-rpath-link,/home/grg/src/tskit/c/build/subprojects/kastore
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:39: multiple definition of `_tmp_file_name'; test_core@exe/tests_test_core.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:39: first defined here
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:40: multiple definition of `_devnull'; test_core@exe/tests_test_core.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:40: first defined here
collect2: error: ld returned 1 exit status
[33/55] Linking target test_stats
FAILED: test_stats
cc  -o test_stats 'test_stats@exe/tests_test_stats.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group libtskit.a subprojects/kastore/libkastore.a libtestlib.a -lm /usr/lib/libcunit.so -Wl,--end-group '-Wl,-rpath,$ORIGIN/:$ORIGIN/subprojects/kastore' -Wl,-rpath-link,/home/grg/src/tskit/c/build/ -Wl,-rpath-link,/home/grg/src/tskit/c/build/subprojects/kastore
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:39: multiple definition of `_tmp_file_name'; test_stats@exe/tests_test_stats.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:39: first defined here
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:40: multiple definition of `_devnull'; test_stats@exe/tests_test_stats.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:40: first defined here
collect2: error: ld returned 1 exit status
[35/55] Linking target test_genotypes
FAILED: test_genotypes
cc  -o test_genotypes 'test_genotypes@exe/tests_test_genotypes.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group libtskit.a subprojects/kastore/libkastore.a libtestlib.a -lm /usr/lib/libcunit.so -Wl,--end-group '-Wl,-rpath,$ORIGIN/:$ORIGIN/subprojects/kastore' -Wl,-rpath-link,/home/grg/src/tskit/c/build/ -Wl,-rpath-link,/home/grg/src/tskit/c/build/subprojects/kastore
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:39: multiple definition of `_tmp_file_name'; test_genotypes@exe/tests_test_genotypes.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:39: first defined here
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:40: multiple definition of `_devnull'; test_genotypes@exe/tests_test_genotypes.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:40: first defined here
collect2: error: ld returned 1 exit status
[36/55] Linking target test_convert
FAILED: test_convert
cc  -o test_convert 'test_convert@exe/tests_test_convert.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group libtskit.a subprojects/kastore/libkastore.a libtestlib.a -lm /usr/lib/libcunit.so -Wl,--end-group '-Wl,-rpath,$ORIGIN/:$ORIGIN/subprojects/kastore' -Wl,-rpath-link,/home/grg/src/tskit/c/build/ -Wl,-rpath-link,/home/grg/src/tskit/c/build/subprojects/kastore
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:39: multiple definition of `_tmp_file_name'; test_convert@exe/tests_test_convert.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:39: first defined here
/usr/bin/ld: libtestlib.a(tests_testlib.c.o):/home/grg/src/tskit/c/build/../tests/testlib.h:40: multiple definition of `_devnull'; test_convert@exe/tests_test_convert.c.o:/home/grg/src/tskit/c/build/../tests/testlib.h:40: first defined here
collect2: error: ld returned 1 exit status
[41/55] Compiling C++ object 'cpp_sorting_example@exe/examples_cpp_sorting_example.cpp.o'
ninja: build stopped: subcommand failed.
```

--- a/c/tests/testlib.c
+++ b/c/tests/testlib.c
@@ -24,6 +24,9 @@
 
 #include "testlib.h"
 
+char *_tmp_file_name;
+FILE *_devnull;
+
 /* Simple single tree example. */
 const char *single_tree_ex_nodes = /*          6          */
     "1  0   -1   -1\n"             /*         / \         */

--- a/c/tests/testlib.h
+++ b/c/tests/testlib.h
@@ -36,8 +36,8 @@
 
 /* Global variables used in the test suite */
 
-char *_tmp_file_name;
-FILE *_devnull;
+extern char *_tmp_file_name;
+extern FILE *_devnull;
 
 int test_main(CU_TestInfo *tests, int argc, char **argv);
 

